### PR TITLE
added login button

### PR DESCRIPTION
**Pull request by lzhou0714**
## Description
For issue #48 
added a login button at the beginning of the home page

## Deployment

- [x] I am making a frontend change, which will be auto-deployed via Heroku.
- [ ] I am making a Ruby change, which will be auto-deployed via Heroku.
  - [ ] I have added one or more gems, and run `rake sorbet:update:all` to generate their types.
  - [ ] I have added or changed a model, and run `rake sorbet:update:all` to generate their types.
- [ ] I am making a database change, which I will manually deploy after my branch is merged.
  - [ ] I have migrated the database using `env RAILS_ENV=production rails db:migrate`.
  - [ ] I have updated the entity relationship diagram via `bundle exec erd`.
  - [ ] I have updated the affected Administrate model dashboards to reflect my DB changes.
- [ ] I am making a Go Lambda change, which I will manually deploy after my branch is merged.
  - [ ] I have deployed the lambdas using `make deploy`.
- [ ] There are other deployment steps I must take after merge, which are:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -92,3 +92,7 @@ public/sitemap*.xml.gz
 # secrets we need for dev and test, so
 # we really don't want this in version control
 .env.*.local
+env.*.local
```
Comment (nathanmsmith) [inline .gitignore:95]: These changes aren't really related to the login button, can we remove them for now? I'm super happy to bring them back in a separate PR, but I'd like to keep each PR _atomic_ (i.e., it only changes/does one thing)

File: Gemfile.lock [hunk 2]
```diff
@@ -2,7 +2,7 @@ GEM
   remote: https://rubygems.org/
   specs:
     Ascii85 (1.1.0)
-    MailchimpMarketing (3.0.78)
+    MailchimpMarketing (3.0.80)
```
Comment (nathanmsmith) [inline Gemfile.lock:5]: Same note here – we don't need to update all the gems in this PR to update the the buttons. Let's leave for a separate PR.

File: db/structure.sql [hunk 3]
```diff
@@ -868,20 +868,6 @@ CREATE SEQUENCE public.pay_webhooks_id_seq
 ALTER SEQUENCE public.pay_webhooks_id_seq OWNED BY public.pay_webhooks.id;
 
 
---
```
Comment (nathanmsmith) [inline db/structure.sql:871]: Same note here – let's save the DB changes for a future PR.

File: app/views/home_page/_call_to_action.html.erb [hunk 4]
```diff
@@ -17,7 +17,11 @@
             Available now. Free for all UCLA students.
           </h2>
           <div class="w-full flex justify-center">
-            <%= button_to 'Sign up', user_google_oauth2_omniauth_authorize_path, data: { turbo: false }, class: 'cursor-pointer mt-6 mx-auto block rounded-md border border-transparent px-5 py-3 bg-gray-900 text-base font-medium text-white shadow hover:bg-black focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-red-500 sm:px-10' %>
+            <%= button_to 'Sign up', user_google_oauth2_omniauth_authorize_path, data: { turbo: false }, class:
```
Comment (nathanmsmith) [inline app/views/home_page/_call_to_action.html.erb:20]: Are we adding a login button here? I just see the sign up button.

**Review by nathanmsmith — CHANGES_REQUESTED** [review]
Woo, thanks @lzhou0714! You'll be the third-ever engineer to contribute to Hotseat :)

Since this is a frontend change, could you add a screenshot or two of what the new buttons look like? I also added a couple of notes below. I'm super happy to discuss further over Zoom or Slack if you have any questions! I'm super excited to get this change in.

A nifty git trick! For the files that aren't related to the button change, you can restore them back to their current versions on the `main` branch with:
```
git checkout main -- .gitignore Gemfile.lock db/structure.sql
```

File: app/views/home_page/_call_to_action.html.erb (hunk 4 above)
Comment (lzhou0714) [reply app/views/home_page/_call_to_action.html.erb:20]: I only added a login button on the header file. I tried putting one here too but it looked a bit strange so I just left it as the sign up button. I can a login button here too if needed. Alternatively, on a different ticket, we could just have the login and sign up button both on a navbar that sticks to the top?

Comment (nathanmsmith) [reply app/views/home_page/_call_to_action.html.erb:20]: Could you post a screenshot of what the login button on the header file looks like?

Comment (lzhou0714) [reply app/views/home_page/_call_to_action.html.erb:20]: Screen Shot 2023-02-04 at 4 42 44 PM
Screen Shot 2023-02-04 at 4 42 38 PM

Comment (nathanmsmith) [reply app/views/home_page/_call_to_action.html.erb]: @lzhou0714 Sweet! Could we just flip the order of the buttons so that "Sign up" is on the right?

And for the call to action part down here, let's update the text to say "Log in / Sign up" - that's what we do on the navbar on other pages.

Comment (lzhou0714) [reply app/views/home_page/_call_to_action.html.erb]: Screen Shot 2023-02-06 at 4 38 14 PM
Screen Shot 2023-02-06 at 4 38 25 PM

Comment (lzhou0714) [reply app/views/home_page/_call_to_action.html.erb]: @nathanmsmith I made the changes! I flipped the color and of login and signup because it looked a little strange but other than that it's basically the same thing just different order.

**Review by nathanmsmith — APPROVED** [review]
Let's ship it! 🚢
